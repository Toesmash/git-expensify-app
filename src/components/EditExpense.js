import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpense extends React.Component {

    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };

    onRemove = () => {
        // console.log(this.props.expense.id)
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/');
    };

    render() {
        return (
            <div>
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                />

                <button
                    onClick={this.onRemove}
                >
                    Remove
                </button>
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch, props) => ({
    startEditExpense: (id, update) => dispatch(startEditExpense(id, update)),
    startRemoveExpense: (expense) => dispatch(startRemoveExpense(expense))
});

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((item) => {
            return item.id === props.match.params.id;
        })
    }
};



export default connect(mapStateToProps, mapDispatchToProps)(EditExpense);