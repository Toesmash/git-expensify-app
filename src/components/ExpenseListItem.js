import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

numeral.register('locale', 'svk', {
    delimiters: {
        thousands: ',',
        decimal: '.'
    },
    abbreviations: {
        thousand: 'k',
        million: 'm',
        billion: 'b',
        trillion: 't'
    },
    currency: {
        symbol: 'â‚¬'
    }
});

// switch between locales
numeral.locale('svk');



const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <Link className='list-item' to={`/edit/${id}`}>
        <div>
            <h3 className='list-item__title'>{description}</h3>
            <span className='list-item__sub-title'>{moment(createdAt).format('DD.MM.YYYY')}</span>
        </div>
        <h3 className='list-item__data'>
            {numeral(amount / 100).format('0,0[.]00$')}
        </h3>
    </Link>
);

export default ExpenseListItem;