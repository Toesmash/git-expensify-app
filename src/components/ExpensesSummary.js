import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expensesTotal, expensesCount }) => {
   return (
      <div className='page-header'>
         <div className='content-container'>
            {
               expensesCount === 1
                  ?
                  <h1 className='page-header__title'>Viewing <span>{expensesCount}</span> expense totaling <span>{numeral(expensesTotal / 100).format('0,0[.]00$')}</span></h1>
                  :
                  <h1 className='page-header__title'>Viewing <span>{expensesCount}</span> expenses totaling <span>{numeral(expensesTotal / 100).format('0,0[.]00$')}</span></h1>
            }
            <div className='page-header__actions'>
               <Link className='button' to="/create">Add Expense</Link>
            </div>
         </div>
      </div>
   );
};

const mapStateToProps = (reduxState) => {
   return {
      expensesTotal: expensesTotal(selectExpenses(reduxState.expenses, reduxState.filters)),
      expensesCount: (selectExpenses(reduxState.expenses, reduxState.filters)).length
   }
};

export default connect(mapStateToProps)(ExpensesSummary);