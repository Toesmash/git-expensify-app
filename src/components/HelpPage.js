import React from 'react';

const HelpPage = () => (
    <div>
        This is HELP text
    </div>
);

export default HelpPage;