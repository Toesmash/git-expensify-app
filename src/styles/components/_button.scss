.button {
   color: white;
   background: $blue;
   padding: $s-size;
   font-size: $font-size-large;
   font-weight: 300;
   border: none;
   line-height: 1;
   display: inline-block;
   text-decoration: none;
}

.button--link {
   background: none;
}

.button--secondary {
   background: #888;
}