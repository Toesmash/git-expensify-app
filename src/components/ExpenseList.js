import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';



export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No expenses</p>
            ): (
                props.expenses.map((item) => (
                    <ExpenseListItem key={item.id} {...item}
                    />
                ))
            )
        }


    </div>
);

const mapStateToProps = (reduxState) => {
    return {
        expenses: selectExpenses(reduxState.expenses, reduxState.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);
