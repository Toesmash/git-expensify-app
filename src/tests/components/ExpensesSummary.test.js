import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should render ExpensesSummary with 1 expense', () => {
   const wrapper = shallow(<ExpensesSummary expensesTotal={345} expensesCount={1}  />);
   expect(wrapper).toMatchSnapshot();
});

test('Should render ExpensesSummary with 0 or multiple expenses', ()=>{
   const wrapper = shallow(<ExpensesSummary expensesTotal={345675} expensesCount={15}/>);
   expect(wrapper).toMatchSnapshot();
});