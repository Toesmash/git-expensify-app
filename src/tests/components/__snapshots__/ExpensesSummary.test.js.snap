// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary with 0 or multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        15
      </span>
       expenses totaling 
      <span>
        3,456.75$
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should render ExpensesSummary with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       expense totaling 
      <span>
        3.45$
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
